src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,112 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
+   * @param status    - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    try {
+      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+
+      Map<String, List<Long>> failedUpdates = new HashMap<>();
+
+      for (Long userId : userIds) {
+        modifyTeacherPendingStatusForUser(userId, requestingUser, status, failedUpdates);
+      }
+
+      if (!failedUpdates.isEmpty()) {
+        String errorMessage = "";
+
+        if (failedUpdates.containsKey(USERS_NOT_FOUND)) {
+          errorMessage = String.format("One or more users could not be found: %s. ",
+              failedUpdates.get(USERS_NOT_FOUND));
+        }
+
+        if (failedUpdates.containsKey(FAILED_TO_SEND)) {
+          errorMessage += String.format("Emails could not be sent to userIds: %s",
+              failedUpdates.get(FAILED_TO_SEND));
+        }
+        return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
+      }
+
+    } catch (NoUserLoggedInException e) {
+      return SegueErrorResponse.getNotLoggedInResponse();
+    } catch (SegueDatabaseException e) {
+      log.error("Database error while trying to change teacher_pending status", e);
+      return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR,
+          "Could not save new teacher_pending status to the database").toResponse();
+    }
+
+    return Response.ok().build();
+  }
+
+  private void modifyTeacherPendingStatusForUser(Long userId, RegisteredUserDTO requestingUser, Boolean status,
+                                                 Map<String, List<Long>> failedUpdates) throws SegueDatabaseException {
+
+    try {
+      RegisteredUserDTO user = this.userManager.getUserDTOById(userId);
+
+      if (null == user) {
+        throw new NoUserException("No user found with this ID.");
+      }
+
+      Boolean oldStatus = user.getTeacherPending();
+      this.userManager.updateTeacherPendingFlag(userId, status);
+      if (log.isInfoEnabled()) {
+        log.info("ADMIN user {} has modified the teacher_pending status of {} [{}] from {} to {}",
+            requestingUser.getEmail(), user.getEmail(), user.getId(), oldStatus, status);
+      }
+      if (Boolean.FALSE.equals(status)) {
```

Comment:
Why is this being compared in this way instead of `if (!status) {`? Is it because we can send `null` values? Is that even allowed? If it's not allowed, perhaps the type needs changing from `Boolean` to `boolean`.

Reply:
This was another sonarlint one - I wrote `!status` initially and it showed `Avoid using boxed "Boolean" types directly in boolean expressions`

Reply:
I think that the right solution depends on whether it's acceptable to pass `null` or not. If the only two right values are either `true` or `false`, then you should use the primitive type (`boolean`, rather than `Boolean`), as that one cannot be `null`.